### PR TITLE
add rest of the adapter

### PR DESCRIPTION
**Pull request by iamsdas**
Signed-off-by: Suryashankar Das <suryashankardas.2002@gmail.com>

**Description**

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by iamsdas** [comment]
cc: @leecalcote @Revolyssup 
btw I don't seem to able to request reviewers

**Comment by Shreyas220** [comment]
@iamsdas can you confirm if cilium is being installed in the cluster

**Comment by leecalcote** [comment]
> cc: @leecalcote @Revolyssup
> btw I don't seem to able to request reviewers

Oh, interesting. Are you able to add/remove labels or any other attributes?

File: internal/config/operations.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	CiliumBookStoreOperation = "cilium_bookstore_app"
+	ServiceName              = "service_name"
+)
+
+func getOperations(op adapter.Operations) adapter.Operations {
+	versions, _ := getLatestReleaseNames(3)
+
+	op[CiliumOperation] = &adapter.Operation{
+		Type:        int32(meshes.OpCategory_INSTALL),
+		Description: "Open Service Mesh",
```
Comment (Revolyssup) [inline internal/config/operations.go:22]: Please remove  "open service mesh" from everywhere

**Review by Revolyssup — CHANGES_REQUESTED** [review]
Please remove the "open service mesh" from  everywhere.

File: internal/config/operations.go (hunk 1 above)
Comment (iamsdas) [reply internal/config/operations.go]: Made the changes 👍

**Comment by iamsdas** [comment]
> > cc: @leecalcote @Revolyssup
> > btw I don't seem to able to request reviewers
> 
> Oh, interesting. Are you able to add/remove labels or any other attributes?

Nope, unable to add/remove anything

**Comment by leecalcote** [comment]
Thanks for mentioning this. I'll get this fixed.

**Comment by leecalcote** [comment]
@iamsdas you should be able to make changes to issues/prs now.

**Review by Revolyssup — CHANGES_REQUESTED** [review]
Please remove definitions and schemas of OSM

**Comment by iamsdas** [comment]
I have made the necessary changes and can confirm that it is running. Currently, I have no idea how to install it and may need some help with it.

File: cilium/error.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+// ErrMeshConfig is the error for mesh config
+func ErrMeshConfig(err error) error {
+	return errors.New(ErrMeshConfigCode, errors.Alert, []string{"Error configuration mesh"}, []string{err.Error(), "Error getting MeshSpecKey config from in-memory configuration"}, []string{}, []string{"Reconnect the adaptor to the meshkit server"})
+}
+
+// ErrRunCiliumCtlCmd is the error for mesh port forward
+func ErrRunCiliumCtlCmd(err error, des string) error {
+	return errors.New(ErrRunCiliumCtlCmdCode, errors.Alert, []string{"Error running istioctl command"}, []string{err.Error()}, []string{"Corrupted istioctl binary", "Command might be invalid"}, []string{})
```
Comment (Revolyssup) [inline cilium/error.go:129]: Remove "istio". Also dont replace with ciliumctl, there is no such thing as cilium ctl. There CLI is just "cilium"

File: internal/config/releases.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+	if limit > len(result) {
+		limit = len(result)
+	}
+
+	return result[:limit], nil
+}
+
+// GetLatestReleases fetches the latest releases from the cilium repository
+func GetLatestReleases(releases uint) ([]*Release, error) {
```
Comment (Revolyssup) [inline internal/config/releases.go:82]: remove this function. We are deprecating its use and will use the one present in meshkit

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: cilium/error.go (hunk 2 above)
Comment (iamsdas) [reply cilium/error.go]: Updated it 👍

File: internal/config/releases.go (hunk 3 above)
Comment (iamsdas) [reply internal/config/releases.go]: Removed the function, also removed the functions which referenced it (I can revert the last commit if that was not what I was supposed to do)

**Comment by leecalcote** [comment]
@iamsdas are you able to run the adapter now?

**Comment by leecalcote** [comment]
@Revolyssup how are we looking? Good to merge?

**Review by Shreyas220 — COMMENTED** [review]
cilium is supposed to be installed in kube-system namespace and currently, on default we are installing it in default namespace 
is there a way we can use kube-system on default // @leecalcote

**Comment by leecalcote** [comment]
@Shreyas220k, yes, see the meshery-istio adapter as an example of how `Istio-system` is specified, not `default`.

File: cilium/install.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	} else {
+		act = mesherykube.INSTALL
+	}
+	return kClient.ApplyHelmChart(mesherykube.ApplyHelmChartConfig{
+		ChartLocation: mesherykube.HelmChartLocation{
+			Repository: repo,
+			Chart:      chart,
+			Version:    version,
+		},
+		Namespace:       namespace,
```
Comment (Shreyas220) [inline cilium/install.go:71]:
```suggestion
		Namespace:       "kube-system",
```

**Review by Shreyas220 — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
Let's get this PR merged in and have a fresh look at what is left to get this adapter initially functional...

// @Revolyssup

**Pull request merged**